File: kraken/src/main/java/glitch/kraken/json/Channel.java
package glitch.kraken.json;

import com.google.gson.annotations.SerializedName;
import glitch.core.api.json.IDObject;
import glitch.core.utils.Immutable;
import java.awt.Color;
import java.util.Locale;
import javax.annotation.Nullable;
import org.immutables.value.Value;

@Immutable
@Value.Immutable
public interface Channel extends IDObject<Long> {
    @Nullable
    Locale getBroadcasterLanguage();

    String getDisplayName();

    Long getFollowers();

    String getGame();

    Locale getLanguage();

    String getLogo();

    boolean isMature();

    @SerializedName(value = "name", alternate = {"username", "login"})
    String getUsername();

    boolean isPartner();

    @Nullable
    String getProfileBanner();

    @Nullable
    Color getProfileBannerBackgroundColor();

    @SerializedName("status")
    String getTitle();

    String getUrl();

    @Nullable
    String getVideoBanner();

    Long getViews();
}

File: kraken/src/main/kotlin/glitch/kraken/object/json/impl/channel.kt
package glitch.kraken.`object`.json.impl

import com.google.gson.annotations.SerializedName
import glitch.api.objects.enums.BroadcasterType
import glitch.kraken.`object`.json.AuthChannel
import glitch.kraken.`object`.json.Channel
import java.awt.Color
import java.time.Instant
import java.util.*

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class ChannelImpl(
        @SerializedName(value = "id", alternate = ["_id"])
        override val id: Long,
        @SerializedName(value = "username", alternate = ["login", "name"])
        override val username: String,
        override val displayName: String,
        override val createdAt: Instant,
        override val updatedAt: Instant,
        override val logo: String,
        override val mature: Boolean,
        @SerializedName("status")
        override val title: String,
        override val broadcasterLanguage: Locale,
        override val game: String,
        override val language: Locale,
        override val partner: Boolean,
        override val videoBanner: String,
        override val profileBanner: String,
        override val profileBannerBackgroundColor: Color,
        override val url: String,
        override val views: Long,
        override val followers: Long
) : Channel

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class AuthChannelImpl(
        @SerializedName(value = "id", alternate = ["_id"])
        override val id: Long,
        @SerializedName(value = "username", alternate = ["login", "name"])
        override val username: String,
        override val displayName: String,
        override val createdAt: Instant,
        override val updatedAt: Instant,
        override val logo: String,
        override val mature: Boolean,
        @SerializedName("status")
        override val title: String,
        override val broadcasterLanguage: Locale,
        override val game: String,
        override val language: Locale,
        override val partner: Boolean,
        override val videoBanner: String,
        override val profileBanner: String,
        override val profileBannerBackgroundColor: Color,
        override val url: String,
        override val views: Long,
        override val followers: Long,
        override val broadcasterType: BroadcasterType,
        override val email: String,
        override val streamKey: String
) : AuthChannel